### PR TITLE
add negative e2e tests for secrets

### PR DESCRIPTION
**Pull request by mukundansundar**
# Description

Added more negative tests for secrets. 

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #1811 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: tests/e2e/secretapp/secretapp_test.go [hunk 1]
```diff
@@ -96,49 +105,55 @@ func generateTestCases() []testCase {
...
+			true,
+			500,
+			"ERR_STATE_GET",
+		},
+		{
+			"get from nonexistent secret store",
+			newRequest(nonexistentStore, utils.SimpleKeyValue{allowedSecret, ""}),
+			newResponse("", utils.SimpleKeyValue{allowedSecret, ""}),
+			true,
+			401,
```
Comment (mukundansundar) [inline tests/e2e/secretapp/secretapp_test.go:155]: handled in #2338

File: tests/e2e/secretapp/secretapp_test.go [hunk 2]
```diff
@@ -96,49 +105,55 @@ func generateTestCases() []testCase {
...
+			403,
+			"PERMISSION_DENIED",
+		},
+		{
+			"get nonexistent secret",
+			newRequest(secretStore, utils.SimpleKeyValue{nonExistentSecret, ""}),
+			newResponse("", utils.SimpleKeyValue{nonExistentSecret, ""}),
+			true,
+			500,
+			"ERR_STATE_GET",
```
Comment (mukundansundar) [inline tests/e2e/secretapp/secretapp_test.go:148]: handed in https://github.com/dapr/dapr/pull/2280/

File: tests/e2e/secretapp/secretapp_test.go [hunk 3]
```diff
@@ -96,49 +105,55 @@ func generateTestCases() []testCase {
...
+			200,
+			"", // no error
+		},
+		{
+			"unallowed secret",
+			newRequest(secretStore, utils.SimpleKeyValue{unallowedSecret, ""}),
+			newResponse("", utils.SimpleKeyValue{unallowedSecret, ""}),
+			true,
+			403,
+			"PERMISSION_DENIED",
```
Comment (wcs1only) [inline tests/e2e/secretapp/secretapp_test.go:140]: Oh that's interesting. Code suggests we expect "ERR_PERMISSION_DENIED" ? If I'm reading the correct line? https://github.com/dapr/dapr/blob/b6ea6f59231c50d2ee054b2a1f5b8d36c83db79c/pkg/http/api.go#L501

Comment (wcs1only) [reply tests/e2e/secretapp/secretapp_test.go:140]: Oh, I see we're using require.Contains().

**Review by wcs1only — APPROVED** [review]

File: tests/e2e/secretapp/secretapp_test.go (hunk 3 above)
Comment (wcs1only) [reply tests/e2e/secretapp/secretapp_test.go:140]: So, it is not essential that you change this, but it wouldn't hurt for consistency sake.

**Review by wcs1only — APPROVED** [review]

**Review by vinayada1 — APPROVED** [review]

**Pull request merged**